### PR TITLE
Fixing siddhi try it tool to hide result column when an exception occurs

### PR DESCRIPTION
**Pull request by thiliA**

### CODE REVIEW

File: components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java [hunk 1]
```diff
@@ -49,195 +49,197 @@
...
-					log.error("Unable to add the datasource" + cds.getDSMInfo().getName(), e);
-				}
-			}
-		} catch (DataSourceException e) {
-			log.error("Unable to populate the data sources in Siddhi engine.", e);
-		}
-	}
+    private static Log log = LogFactory.getLog(SiddhiTryItClient.class);
+    private static Gson gson = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
+    private String errMsg = "";
```
Comment (ramindu90) [inline components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java]: Lets remove the unnecessary initialization with the empty string.

Comment (thiliA) [inline components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java]: Sure, I have fixed it in PR:https://github.com/wso2/carbon-event-processing/pull/179

File: components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java [hunk 2]
```diff
@@ -49,195 +49,197 @@
...
+        long timeStamp = date.getTime();
+        return timeStamp;
+    }
+
+    /**
+     * Load data sources configuration
+     *
+     * @param siddhiManager SiddhiManager object
+     */
+    public static void loadDataSourceConfiguration(SiddhiManager siddhiManager) {
```
Comment (sajithshn) [inline components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java]: Shouldn't we throw an exception in this method? If something goes wrong here, processData() will continue to do rest of the work assuming that the data source configurations are loaded.

Comment (thiliA) [inline components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java]: @sajithshn 
Thank you for pointing it out. Fixed in PR: #180

File: components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java [hunk 3]
```diff
@@ -49,195 +49,197 @@
...
+                    }
+                } else if (delayPatternMatcher.find()) {
+                    Thread.sleep(Long.parseLong(delayPatternMatcher.group(2)));
+                } else {
+                    if (!inputStreamEventArray[i].equals("")) {
+                        errMsg = "Error in event \"  " +
+                                inputStreamEventArray[i] +
+                                "\n\"." +
+                                " Expected format: &lt;eventStreamName&gt;=[&lt;attribute1&gt;,&lt;attribute2&gt;]";
+                        throw new Exception(errMsg);
```
Comment (prabathabey) [inline components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java]: Throwing generic exceptions needs to be avoided.

File: components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java [hunk 4]
```diff
@@ -49,195 +49,197 @@
...
+                    }
+                }
+            }
+            Thread.sleep(500);
+            return map;
+        } catch (Throwable e) {
+            //catch throwable since there could be execution plan validation exceptions
+            errMsg = "Error occurred while processing. " + e.getMessage();
+            log.error(errMsg, e);
+            throw new Exception(errMsg, e);
```
Comment (prabathabey) [inline components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java]: Same as above.

File: components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java [hunk 5]
```diff
@@ -49,195 +49,197 @@
...
+     */
+    private long createTimeStamp(String dateTime) throws Exception {
+        SimpleDateFormat dateFormatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+        Date date;
+        try {
+            date = dateFormatter.parse(dateTime);
+        } catch (Exception e) {
+            errMsg = "Error occurred while parsing date " + e.getMessage();
+            log.error(errMsg, e);
+            throw new Exception(errMsg, e);
```
Comment (prabathabey) [inline components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java]: Same as above.

File: components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java [hunk 6]
```diff
@@ -49,195 +49,197 @@
...
+            if (tenantId > -1) {
+                DataSourceManager.getInstance().initTenant(tenantId);
+            }
+            List<CarbonDataSource> dataSources = SiddhiTryItValueHolder.getDataSourceService().getAllDataSources();
+            for (CarbonDataSource cds : dataSources) {
+                try {
+                    if (cds.getDSObject() instanceof DataSource) {
+                        siddhiManager.setDataSource(cds.getDSMInfo().getName(), (DataSource) cds.getDSObject());
+                    }
+                } catch (Exception e) {
```
Comment (prabathabey) [inline components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java]: Catching generic exceptions too is not encouraged.

File: components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java (hunk 3 above)
Comment (thiliA) [inline components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java]: @prabathabey
Thank you for pointing out. Fixed in PR: #181

File: components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java (hunk 5 above)
Comment (thiliA) [inline components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java]: @prabathabey
Thank you for pointing out. Fixed in PR: #181

File: components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java (hunk 6 above)
Comment (thiliA) [inline components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java]: @prabathabey
Thank you for pointing out. Fixed in PR: #181

File: components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java (hunk 4 above)
Comment (thiliA) [inline components/siddhi-tryit/org.wso2.carbon.siddhi.tryit.ui/src/main/java/org/wso2/carbon/siddhi/tryit/ui/SiddhiTryItClient.java]: @prabathabey 
Thank you for pointing out. We have not created any component specific exceptions as this is only a ui component. As for now, Siddhi try it ui component escalates the error message to front UI (directly called by a jsp file). So IMHO, I believe the above approach will be okay. Please correct me if I am wrong.

**Pull request merged**